Data/Migrations/PuzzleServerContextModelSnapshot.cs

```diff
@@ -799,13 +799,15 @@ protected override void BuildModel(ModelBuilder modelBuilder)
 
             modelBuilder.Entity("ServerCore.DataModel.PuzzleAuthors", b =>
                 {
-                    b.HasOne("ServerCore.DataModel.Puzzle", "Puzzle")
+                    b.HasOne("ServerCore.DataModel.PuzzleUser", "Author")
```

Comment:
If you add [Required] to the Puzzle and Author properties, you probably don't need the customization in this method.